modules/urlsigning-service-impl/src/main/java/org/opencastproject/security/urlsigning/provider/impl/AbstractUrlSigningProvider.java

```diff
@@ -75,32 +80,14 @@
   /**
    * A class to contain the necessary key entries for url signing.
    */
-  private class KeyEntry {
-    private final String keyId;
-    private final String key;
-    private final String organization;
-
-    KeyEntry(String keyId, String key, String organization) {
-      this.keyId = keyId;
-      this.key = key;
-      this.organization = organization;
-    }
-
-    public String getId() {
-      return keyId;
-    }
-
-    public String getKey() {
-      return key;
-    }
-
-    public String getOrganization() {
-      return organization;
-    }
+  private static class KeyEntry {
+    private String key = null;
+    private String url = null;
+    private String organization = ANY_ORGANIZATION;
   }
 
   /** The map to contain the list of keys, their ids and the urls they match. */
-  private Map<String, KeyEntry> keys = new TreeMap<>();
+  private Map<String, KeyEntry> keys = new HashMap<>();
```

Comment:
@JulianKniephoff Looking at the lines 

https://github.com/opencast/opencast/pull/731/files#diff-34839f84f847d48880db263f855ba1c9R85

and

https://github.com/opencast/opencast/pull/731/files#diff-34839f84f847d48880db263f855ba1c9R119

it seems that either me or you have misunderstood the purpose of "keys".

To my understanding, the URL signing provider needs fast lookup of keys based on the URL. That's why they used a TreeMap with Map<String(=URL),KeyEntry>.

If that is correct, you have "un-"optimized that code and made it slower ;-)